test/fw/ptl/lib/ptl_server.py

```diff
@@ -0,0 +1,6087 @@
+
+    def set_connect_timeout(self, timeout=0):
+        """
+        Set server connection timeout
+
+        :param timeout: Timeout value
+        :type timeout: int
+        """
+        self._conn_timeout = timeout
+
+    def get_op_mode(self):
+        """
+        Returns operating mode for calls to the PBS server.
+        Currently, two modes are supported, either the ``API``
+        or the ``CLI``. Default is ``API``
+        """
+        if (not API_OK or (self.ptl_conf['mode'] == PTL_CLI)):
+            return PTL_CLI
+        return PTL_API
+
+    def set_op_mode(self, mode):
+        """
+        set operating mode to one of either ``PTL_CLI`` or
+        ``PTL_API``.Returns the mode that was set which can
+        be different from the value requested, for example, if
+        requesting to set ``PTL_API``, in the absence of the
+        appropriate SWIG wrappers, the library will fall back to
+        ``CLI``, or if requesting ``PTL_CLI`` and there is no
+        ``PBS_EXEC`` on the system, None is returned.
+
+        :param mode: Operating mode
+        :type mode: str
+        """
+        if mode == PTL_API:
+            if self._conn is not None or self._conn < 0:
+                self._conn = None
+            if not API_OK:
+                self.logger.error(self.logprefix +
+                                  'API submission is not available')
+                return PTL_CLI
+        elif mode == PTL_CLI:
+            if ((not self.has_snap) and
+                not os.path.isdir(os.path.join(self.client_conf['PBS_EXEC'],
+                                               'bin'))):
+                self.logger.error(self.logprefix +
+                                  'PBS commands are not available')
+                return None
+        else:
+            self.logger.error(self.logprefix + "Unrecognized operating mode")
+            return None
+
+        self.ptl_conf['mode'] = mode
+        self.logger.info(self.logprefix + 'server operating mode set to ' +
+                         mode)
+        return mode
+
+    def add_expect_action(self, name=None, action=None):
+        """
+        Add an action handler to expect. Expect Actions are
+        custom handlers that are triggered when an unexpected
+        value is encountered
+
+        :param name: Action name
+        :type name: str or None
+        :param action: Action to add
+        """
+        if name is None and action.name is None:
+            return
+        if name is None and action.name is not None:
+            name = action.name
+
+        if not self.actions.has_action(name):
+            self.actions.add_action(action, self.shortname)
+
+    def set_attributes(self, a={}):
+        """
+        set server attributes
+
+        :param a: Attribute dictionary
+        :type a: Dictionary
+        """
+        super(Server, self).set_attributes(a)
+        self.__dict__.update(a)
+
+    def isUp(self, inst, max_attempts=None):
```

Comment:
Do we still need inst parameter ? we have separate services and isUp method is available in all.